src/Steps/Release/MergeUpRelease.php

```diff
@@ -0,0 +1,201 @@
+        bool $pushToRemote
+    ) {
+        $currentBranch = $releasePlanNode->getLibrary()->getBranch();
+
+        // Skip upgrade-only modules and anything not on an actual branch (e.g. using a pre-existing tag)
+        if ($releasePlanNode->getLibrary()->isUpgradeOnly() || !$currentBranch) {
+            return;
+        }
+        $currentLibrary = $releasePlanNode->getLibrary()->getName();
+
+        // Skip if it's not a proper release branch.
+        if (!$releasePlanNode->isOnReleaseBranch()) {
+            $this->log(
+                $output,
+                "Library $currentLibrary is on a non-release branch '$currentBranch'"
+                    . "Please checkout the release branch and then run the command again.",
+                'error'
+            );
+            die();
+        }
+
+        // Do the merge up recursively
+        foreach ($releasePlanNode->getItems() as $item) {
+            $this->mergeRecursive($output, $item, $pushToRemote);
+        }
+        $this->mergeLibrary($output, $releasePlanNode, $pushToRemote);
+    }
+
+    /**
+     * Performs a merge up for a single library
+     *
+     * @param OutputInterface $output
+     * @param LibraryRelease $releasePlanNode Node in release plan being released
+     */
+    protected function mergeLibrary(OutputInterface $output, LibraryRelease $releasePlanNode, bool $pushToRemote)
+    {
+        $library = $releasePlanNode->getLibrary();
+        $releaseBranch = $library->getBranch();
+        $minorBranch = str_replace(LibraryRelease::RELEASE_BRANCH_SUFFIX, '', $releaseBranch);
+        $majorBranch = $releasePlanNode->getVersion()->getMajor();
+
+        // Merge up to minor branch, then to major branch
+        $this->mergeUp($output, $releasePlanNode, $releaseBranch, $releaseBranch, $minorBranch, $pushToRemote, false);
+        $this->mergeUp($output, $releasePlanNode, $releaseBranch, $minorBranch, $majorBranch, $pushToRemote, true);
+
+        // Always check out the release branch at the end
+        $library->checkout($output, $releaseBranch);
+    }
+
+    /**
+     * Merge up from one branch to another, setting the appropriate constraints for released dependencies as we go.
+     */
+    protected function mergeUp(
+        OutputInterface $output,
+        LibraryRelease $releasePlanNode,
+        string $releaseBranchName,
+        string $mergeFrom,
+        string $mergeInto,
+        bool $pushToRemote,
+        bool $isMajorBranch
+    ) {
+        $library = $releasePlanNode->getLibrary();
+        $name = $library->getName();
+
+        // Step 1. Checkout branch to merge into
+        $library->checkout($output, $mergeInto);
+
+        $this->log(
+            $output,
+            "Merging library <info>{$name}</info> from <info>{$mergeFrom}</info> to <info>{$mergeInto}</info>"
+        );
+
+        // Step 2. Merge up if there's anything to merge
+        if ($library->hasDiff($output, $mergeFrom)) {
```

Comment:
`$mergeFrom` is the release branch (which we _know_ exists because we've already checked we're on that branch) the first time and then the minor branch (which we _also_ know exists because we just checked it out and merged into it) the second time.

So you won't ever actually be in the situation that `$mergeFrom` doesn't exist by the time you hit this - but if you did, a `ProcessException` would be thrown from the git library with a message like "fatal: ambiguous argument 'some-branch': unknown revision or path not in the working tree."